src/use_case/settings/SettingsInteractor.java

```diff
@@ -1,25 +1,34 @@
 package use_case.settings;
 
-import entities.auth.AbstractUser;
-
 public class SettingsInteractor implements SettingsInputBoundary {
   // TODO: Add this after other code is merged
   // final SettingsUserDataAccessInterface settingsUserDataAccessInterface;
   final SettingsOutputBoundary settingsPresenter;
 
+  final UserSettingsDataAccessInterface userSettingsDataAccessObject;
+  final RoomsSettingsDataAccessInterface roomsSettingsDataAccessObject;
+
   public SettingsInteractor(
-      // SettingsUserDataAccessInterface settingsDataAccessInterface,
+      UserSettingsDataAccessInterface userSettingsDataAccessInterface,
+      RoomsSettingsDataAccessInterface roomsSettingsDataAccessInterface,
       SettingsOutputBoundary settingsOutputBoundary) {
-    // this.settingsDataAccessObject = settingsDataAccessInterface;
+    this.userSettingsDataAccessObject = userSettingsDataAccessInterface;
+    this.roomsSettingsDataAccessObject = roomsSettingsDataAccessInterface;
     this.settingsPresenter = settingsOutputBoundary;
   }
 
   @Override
-  public void execute(SettingsInputData userInputData) {
-    AbstractUser user = userInputData.getUser();
-
-    // TODO: Write interaction after other code is merged
-    SettingsOutputData settingsOutputData = new SettingsOutputData(user, true);
-    settingsPresenter.prepareSuccessView(settingsOutputData);
+  public void executeChangeUsername(SettingsInputData settingsInputData) {
+    try {
+      userSettingsDataAccessObject.changeDisplayName(
+          settingsInputData.getUser(), settingsInputData.getUpdatedUsername());
+      roomsSettingsDataAccessObject.propogateDisplayNameChange(
+          settingsInputData.getUser(), settingsInputData.getUpdatedUsername());
+      SettingsOutputData settingsOutputData = new SettingsOutputData(null, false);
```

Comment:
New username should be passed here to update the view instead of having to login and logout again.